src/extension/background-script/connectors/kollider.ts

```diff
@@ -0,0 +1,363 @@
+        })
+      )
+      .sort((a, b) => {
+        return b.settleDate - a.settleDate;
+      });
+
+    return {
+      data: {
+        invoices,
+      },
+    };
+  }
+
+  async getInfo(): Promise<GetInfoResponse> {
+    return { data: { alias: `Kollider ${this.currency}` } };
+  }
+
+  async getBalance(): Promise<GetBalanceResponse> {
+    await this.loadAccounts();
+    const account = await this.findAccount(this.currency);
+
+    if (!account) {
+      throw new Error("Account not found");
+    }
+
+    const balance = parseFloat(account.balance);
+
+    return {
+      data: {
+        balance,
+        currency: account.currency,
+      },
+    };
+  }
+
+  async sendPayment(args: SendPaymentArgs): Promise<SendPaymentResponse> {
+    const data = await this.request<{
+      error?: string;
+      success: boolean;
+      payment_hash: string;
+      uid: string;
+      currency: ACCOUNT_CURRENCIES;
+      payment_request: string;
+      amount: string;
+      fees: string;
+      rate: string;
+    }>("POST", "/payinvoice", {
+      payment_request: args.paymentRequest,
+      currency: this.config.currency,
+    });
+    if (data.error) {
+      throw new Error(data.error);
+    }
+    const payment_route = {
+      total_amt: parseFloat(data.amount),
+      total_fees: parseFloat(data.fees),
+    };
+    return {
+      data: {
+        preimage: "", // data.payment_preimage as string,
+        paymentHash: data.payment_hash,
+        route: payment_route,
+      },
+    };
+  }
+
+  async keysend(args: KeysendArgs): Promise<SendPaymentResponse> {
+    throw new Error("Not yet supported with the currently used account.");
+  }
+
+  async checkPayment(args: CheckPaymentArgs): Promise<CheckPaymentResponse> {
+    throw new Error("Not yet supported with the currently used account.");
+  }
+
+  signMessage(args: SignMessageArgs): Promise<SignMessageResponse> {
+    // make sure we got the config to create a new key
+    if (!this.config.username || !this.config.password) {
+      return Promise.reject(new Error("Missing config"));
+    }
+    if (!args.message) {
+      return Promise.reject(new Error("Invalid message"));
+    }
+    const message = sha256(args.message).toString(Hex);
+    const keyHex = sha256(
+      `kollider://${this.config.username}:${this.config.password}`
+    ).toString(Hex);
+    if (!keyHex) {
+      return Promise.reject(new Error("Could not create key"));
+    }
+    const signer = new HashKeySigner(keyHex);
+    const signedMessageDERHex = signer.sign(message).toDER("hex");
+    // make sure we got some signed message
+    if (!signedMessageDERHex) {
+      return Promise.reject(new Error("Signing failed"));
+    }
+    return Promise.resolve({
+      data: {
+        message: args.message,
+        signature: signedMessageDERHex,
+      },
+    });
+  }
+
+  async makeInvoice(args: MakeInvoiceArgs): Promise<MakeInvoiceResponse> {
+    const amount = parseFloat(args.amount.toString()) / 100000000;
```

Comment:
You could use this?:  
https://github.com/getAlby/lightning-browser-extension/blob/refactor/manifest-v3-support/src/extension/background-script/actions/cache/getCurrencyRate.ts#L18

Reply:
👍